rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
Set aside the fact that it's only called from one thread, I still cannot see the reason that this initialization can be called several times from different threads. Using TLS won't ensure the thread safety at all.